picard/file.py

```diff
@@ -400,35 +400,47 @@ def _save_images(self, dirname, metadata):
             image.save(dirname, metadata, counters)
 
     def _move_additional_files(self, old_filename, new_filename):
-        """Move extra files, like playlists..."""
-        old_path = os.path.dirname(old_filename)
-        new_path = os.path.dirname(new_filename)
-        try:
-            names = set(os.listdir(old_path))
-        except os.error:
-            log.error("Error: {} directory not found".naming_format(old_path))
-            return
-        filtered_names = {name for name in names if name[0] != "."}
-        for pattern in config.setting["move_additional_files_pattern"].split():
+        """Move extra files, like images, playlists..."""
+        patterns = config.setting["move_additional_files_pattern"]
+        pattern_regexes = set()
+        for pattern in patterns.split():
             pattern = pattern.strip()
             if not pattern:
                 continue
             pattern_regex = re.compile(fnmatch.translate(pattern), re.IGNORECASE)
-            file_names = names
-            if pattern[0] != '.':
-                file_names = filtered_names
-            for old_file in set(file_names):
-                if pattern_regex.match(old_file):
-                    names.discard(old_file)
-                    filtered_names.discard(old_file)
-                    new_file = os.path.join(new_path, old_file)
-                    old_file = os.path.join(old_path, old_file)
-                    # FIXME we shouldn't do this from a thread!
-                    if self.tagger.files.get(decode_filename(old_file)):
-                        log.debug("File loaded in the tagger, not moving %r", old_file)
+            match_hidden = pattern.startswith('.')
+            pattern_regexes.add((pattern_regex, match_hidden))
+        if not pattern_regexes:
+            return
+        new_path = os.path.dirname(new_filename)
+        old_path = os.path.dirname(old_filename)
+        moves = set()
+        try:
+            # TODO: use with statement with python 3.6+
```

Comment:
We could look whether we want to increase the minimum version. I think it depends a bit which distros we want to support. But e.g. the oldest Ubuntu we currently can build for is Ubuntu 18.04 Bionic, and that has Python 3.6 (in theory we could also build on 17.10, but that is no longer supported).